hera_cal/redcal.py

```diff
@@ -789,3 +837,335 @@ def is_redundantly_calibratable(antpos, bl_error_tol=1.0):
+    nInt_to_load integrations at a time and skipping and flagging times when the sun is above solar_horizon.
+    
+    Arguments:
+        input_data: path to uvh5 visibility data file to calibrate.
+        firstcal_suffix: string to append to input_data path for firstcal calfits file to save
+        omnical_suffix: string to append to input_data path for omnical calfits file to save
+        omnivis_suffix: string to append to input_data path for omnical visibility solutions to save as uvh5
+        outdir: folder to save data products. If None, will be the same as the folder containing input_data
+        ant_metrics_file: path to file containing ant_metrics readable by hera_qm.metrics_io.load_metric_file.
+            Used for finding ex_ants and is combined with antennas excluded via ex_ants.
+        clobber: if True, overwrites existing files for the firstcal and omnical results
+        nInt_to_load: number of integrations to load and calibrate simultaneously. Lower numbers save memory,
+            but incur a CPU overhead.
+        pol_mode: polarization mode of redundancies. Can be '1pol', '2pol', '4pol', or '4pol_minV'.
+            See recal.get_reds for more information.
+        ex_ants: list of antennas to exclude from calibration and flag. Can be either antenna numbers or
+            antenna-polarization tuples. In the former case, all pols for an antenna will be excluded.
+        ant_z_thresh: threshold of modified z-score (like number of sigmas but with medians) for chi^2 per 
+            antenna above which antennas are thrown away and calibration is re-run iteratively. Z-scores are
+            computed independently for each antenna polarization, but either polarization being excluded
+            triggers the entire antenna to get flagged (when multiple polarizations are calibrated)
+        solar_horizon: float, Solar altitude flagging threshold [degrees]. When the Sun is above
+            this altitude, calibration is skipped and the integrations are flagged.
+        conv_crit: maximum allowed relative change in omnical solutions for convergence
+        maxiter: maximum number of omnical iterations allowed before it gives up
+        check_every: compute omnical convergence every Nth iteration (saves computation).
+        check_after: start computing omnical convergence only after N iterations (saves computation).
+        gain: The fractional step made toward the new solution each omnical iteration. Values in the
+            range 0.1 to 0.5 are generally safe. Increasing values trade speed for stability.
+        append_to_history: string to add to history of output firstcal and omnical files
+        verbose: print calibration progress updates
+
+    Returns:
+        cal: the dictionary result of the final run of redcal_partial_io_iteration (see above for details)
+    '''
+    assert isinstance(input_data, str), 'input_data must be a single string path to a uvh5 visibility data file.'
```

Comment:
Why not, optionally, and hd object as well, in case you already have it open.

Reply:
This mucks with how we do all the file suffixes. I think we want to re-think how they are handled anyway, so perhaps you can make an issue and we can put on this for now?

Reply:
Ok, nevermind. This is done now.